### DISCUSSION TITLE
does need osrm-extract after any osrm-customization?

### DISCUSSION THREAD

**Discussion by mojtaba-roohi** [Project-OSRM/osrm-backend#6907] [General]
hello
I have two question
which needs to be extracted after each customization?
does osrm-route not respond to routing requests when customizing?

**Comment by DennisOSRM** [comment]
No, osrm-extract does not need to be re-run after customization. The MLD pipeline has three stages with different triggers:
	∙	osrm-extract — only when your OSM input data changes
	∙	osrm-partition — only after a new extract (graph topology changed)
	∙	osrm-customize — as often as you like (e.g. for live traffic updates); this is specifically designed to be the fast, repeatable step
osrm-routed keeps serving requests while customization runs, using the previously loaded data. Once osrm-customize finishes, send SIGUSR1 to reload. For zero-downtime updates, use a blue/green setup and switch your load balancer once the new instance is ready.
